command-parser.js

```diff
@@ -113,6 +113,21 @@ class CommandContext {
 		this.inputUsername = '';
 	}
 
+	checkFormat(room, message) {
+		if (!room || !room.stretching && !room.capitals) return false;
+		let formatError = false;
+		let stretchMatch = message.match(/(.+?)\1{7,}/i);
+		let capsMatch = message.match(/[A-Z\s]{18,}/);
+		if (room.stretching && stretchMatch) {
+			formatError = "too much stretching.";
+		}
+		if (room.capitals && message.match(/[A-Z\s]{18,}/)) {
+			formatError = "too many capital letters.";
+		}
+		if (room.stretching && room.capitals && stretchMatch && capsMatch) formatError = "too much stretching and too many capital letters.";
```

Comment:
Also you might as well move `room.stretching && ` into the `stretchMatch = ` definition. Save some repetition and makes for more readable code and slightly improves performance.